synapse/tools/sync_020.py

```diff
@@ -0,0 +1,651 @@
+
+    async def _destIterLyrNodeedits(self, prx, queue, lyriden):
+        '''
+        Batch available source splices in a queue as nodeedits and push to the destination layer proxy.
+        Nodeedit boundaries are defined by the ndef and prov iden.
+        Will run as long as queue is not fini'd.
+
+        Args:
+            prx (s_telepath.Proxy): Proxy to destination layer
+            queue (s_queue.Window): Layer queue for splices
+            lyriden (str): Layer iden
+        '''
+        fair_iter = self.push_fair_iter
+        err_lim = self.err_lim
+
+        ndef = None
+        prov = None
+        nodesplices = []
+        nodespliceoffs = []
+
+        cnt = 0
+        errs = 0
+        async for offs, splice in queue:
+            queuelen = len(queue.linklist)
+            next_ndef = splice[1]['ndef']
+            next_prov = splice[1].get('prov')
+
+            # current splice is a new node or has new prov iden or the queue is empty
+            # so create prior node nodeedit and push to destination layer
+            if ndef is not None and (next_ndef != ndef or (prov is not None and next_prov != prov) or queuelen == 0):
+                err, ne, meta = None, None, None
+
+                try:
+                    err, ne, meta = await self._trnNodeSplicesToNodeedit(ndef, nodesplices)
+                    if err is None:
+                        await prx.storNodeEditsNoLift([ne], meta)
+                        self.push_offs.set(lyriden, offs + 1)
+
+                except asyncio.CancelledError:  # pragma: no cover
+                    raise
+                except (ConnectionError, s_exc.IsFini):
+                    # put back last and nodesplices
+                    queue.linklist.appendleft((offs, splice))
+                    qadd = zip(reversed(nodespliceoffs), reversed(nodesplices))
+                    queue.linklist.extendleft(qadd)
+                    raise
+                except Exception as e:
+                    err = {'mesg': s_common.excinfo(e), 'splices': nodesplices, 'nodeedits': ne, 'meta': meta}
+                    logger.warning(err['mesg'])
+
+                if err is not None:
+                    errs += 1
+                    await self._setLyrErr(lyriden, offs, err)
+                    if errs >= err_lim:
+                        raise SyncErrLimReached(mesg='Error limit reached - correct or increase err_lim to continue')
+
+                nodesplices = []
+                nodespliceoffs = []
+
+            ndef = next_ndef
+            prov = next_prov
+            nodesplices.append(splice)
+            nodespliceoffs.append(offs)
+
+            cnt += 1
+
+            if queuelen % 10000 == 0:
+                logger.debug(f'{lyriden} queue reader status: read={cnt}, errs={errs}, size={queuelen}')
+
+            if queuelen == 0 or cnt % fair_iter == 0:
+                await asyncio.sleep(0)
+
+def getParser():
+    https = os.getenv('SYN_UNIV_HTTPS', '4443')
```

Comment:
What is the significance of the name SYN_UNIV here?

Reply:
copypasta from our other services that I assumed was used in some deployment configuration that leveraged env vars, but I can remove it and set the defaults directly.